### PR TITLE
feat(source): add default tag and set default value

### PR DESCRIPTION
**Pull request by yejiayu**
**What this PR does / why we need it**:
- Add `default` source tag. 
- All basic types of params have a default value.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @kdada 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: service/source.go [hunk 1]
```diff
@@ -385,15 +385,18 @@ func (g *AutoParameterGenerator) Validate(name string, defaultValue interface{},
 	return err
 }
 
-func (g *AutoParameterGenerator) split(tag string) (source string, name string, err error) {
+func (g *AutoParameterGenerator) split(tag string) (source, name, defaultValue string, err error) {
 	result := strings.Split(tag, ",")
+
 	if len(result) == 1 {
```
Comment (kdada) [inline service/source.go]: Use `switch`.

File: service/source.go [hunk 2]
```diff
@@ -385,15 +385,18 @@ func (g *AutoParameterGenerator) Validate(name string, defaultValue interface{},
...
-		return result[0], "", nil
-	}
-	if len(result) != 2 {
-		return "", "", invalidFieldTag.Error(tag)
+		return strings.TrimSpace(result[0]), "", "", nil
+	} else if len(result) == 2 {
+		return strings.TrimSpace(result[0]), strings.TrimSpace(result[1]), "", nil
+	} else if len(result) == 3 {
+		return strings.TrimSpace(result[0]), strings.TrimSpace(result[1]), strings.TrimSpace(result[2]), nil
```
Comment (kdada) [inline service/source.go]: Why don't you use the format: `source,name,default=defaultValue` ?

File: service/source.go [hunk 3]
```diff
@@ -427,6 +430,23 @@ func (g *AutoParameterGenerator) generate(ctx context.Context, vc ValueContainer
 		if err != nil {
 			return err
 		}
+		if ins == nil {
+			var value interface{}
+			if defaultValue != "" {
+				var err error
+				if c := ConverterFor(field.Type); c != nil {
```
Comment (kdada) [inline service/source.go]: What if `c == nil`?

File: service/source.go [hunk 4]
```diff
@@ -448,3 +468,20 @@ func (g *AutoParameterGenerator) enum(index []int, typ reflect.Type, f func(inde
 	}
 	return nil
 }
+
+func zero(t reflect.Type) interface{} {
+	switch t.Kind() {
+	case reflect.Bool:
+		return false
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return 0
```
Comment (kdada) [inline service/source.go]: It's wrong. You can't treat `int` as `uint`, `uint8` and etc. 
All values should have its own type rather `int`.

File: service/source.go [hunk 5]
```diff
@@ -448,3 +468,20 @@ func (g *AutoParameterGenerator) enum(index []int, typ reflect.Type, f func(inde
...
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return 0
```
Comment (kdada) [inline service/source.go]: Ditto.

File: service/source.go [hunk 6]
```diff
@@ -448,3 +468,20 @@ func (g *AutoParameterGenerator) enum(index []int, typ reflect.Type, f func(inde
...
+	case reflect.Float32, reflect.Float64:
+		return 0
```
Comment (kdada) [inline service/source.go]: Ditto.

File: service/source.go (hunk 3 above)
Comment (yejiayu) [reply service/source.go]: ins == value(nil)

File: service/source.go [hunk 7]
```diff
@@ -385,15 +385,37 @@ func (g *AutoParameterGenerator) Validate(name string, defaultValue interface{},
...
+	switch len(result) {
+	case 1:
+		source = strings.TrimSpace(result[0])
+	case 2:
+		source = strings.TrimSpace(result[0])
+		name = strings.TrimSpace(result[1])
+	case 3:
+		source = strings.TrimSpace(result[0])
+		name = strings.TrimSpace(result[1])
+		params := strings.Split(strings.TrimSpace(result[2]), ";")
```
Comment (kdada) [inline service/source.go]: `path,application,default=123,anotherKey=anotherValue` is more preferable than `Path,application,default=123;anotherKey=anotherValue`.

File: service/executor.go [hunk 8]
```diff
@@ -361,8 +361,12 @@ func (e *executor) Execute(ctx context.Context) (err error) {
 		if err != nil {
 			return writeError(ctx, e.producers, err)
 		}
-		if result == nil && p.defaultValue != nil {
-			result = p.defaultValue
+		if result == nil {
+			if p.defaultValue != nil {
+				result = p.defaultValue
+			} else {
+				result = zero(p.targetType)
```
Comment (kdada) [inline service/executor.go]: You must generate a completely correct zero value here.

**Comment by yejiayu** [comment]
ping @kdada

**Pull request merged**